### ISSUE TITLE
Can we programatically determine which assignments count for slip days?

### ISSUE THREAD

**Issue by cycomachead** [cs10/Alonzo#91]
We can use bCourses assignment groups. We probably can't do exact maching on assignment titles, since that's not consistent. We could search for text in the description, but that reveals too much to students?

Do we care if students know slip days aren't counted for certain things?

**Comment by aaaschmitt** [comment]
What about point value of the assignment? Idk why I just thought of this.....

**Comment by cycomachead** [comment]
What do you mean?

The points are included in the assignment response, as well as the graded points if the assignment is graded.

**Comment by cycomachead** [comment]
The issue is if we can really codify the slip days assignment deductions...which actually isn't too hard.

**Comment by aaaschmitt** [comment]
Oh I thought this was just about how can we tell if an assignment should be used for slip days. I was wondering if we could use the number of points that an assignment is worth to tell us this. Like for example all assignments worth more than X number of points are eligible for slip day calculations.

**Comment by cycomachead** [comment]
I initially meant just as a way of fixing the slip day tracker, but yeah, the second thing would be great too.
I'd like to not need to grab assignment ID's at the beginning of each semester. 
So the list will look like: "Homework 1", ... "Final Project" and such.

The first part just requires some fuzzy matching in the canvas LMS package, which will be useful anyway...I should do that soon.

The second part is a more messy algorithm, but should be doable.
The proper way (I think) to do this is to just take all late assignments, and sort by min score, and then start subtracting points for when people are over 4 days.

**Comment by aaaschmitt** [comment]
Oh so like fully automating the end of year grading process. That would be cool.

**Comment by cycomachead** [comment]
Yeah, I think everything we could do to make this simpler would be great.

Like once we verify deductions we could automate updating scores and posting comments on bCourses.
